src/main/java/com/zerobase/together/service/UserService.java

```diff
@@ -0,0 +1,113 @@
+package com.zerobase.together.service;
+
+import com.zerobase.together.model.Auth;
+import com.zerobase.together.model.Auth.SignUpWithPartner;
+import com.zerobase.together.persist.CoupleRepository;
+import com.zerobase.together.persist.UserRepository;
+import com.zerobase.together.persist.entity.CoupleEntity;
+import com.zerobase.together.persist.entity.UserEntity;
+import java.time.LocalDateTime;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class UserService implements UserDetailsService {
+
+  private final PasswordEncoder passwordEncoder;
+  private final CoupleRepository coupleRepository;
+  private final UserRepository userRepository;
+
+  /**
+   * userId로 회원 정보를 불러오는 메서드
+   *
+   * @param userId
+   * @return 회원 정보
+   * @throws UsernameNotFoundException
+   */
+  @Override
+  public UserDetails loadUserByUsername(String userId) throws UsernameNotFoundException {
+    return this.userRepository.findByUserId(userId)
+        .orElseThrow(() -> new UsernameNotFoundException("유저가 존재하지 않습니다. -> " + userId));
+  }
+
+  /**
+   * 커플 중 처음 회원 등록하는 메서드
+   *
+   * @param user (회원에 대한 정보)
+   * @return 저장된 회원 정보
+   */
+  public UserEntity register(Auth.SignUp user) {
+    boolean exists = this.userRepository.existsByUserId(user.getUserId());
+    if (exists) {
+      throw new RuntimeException("이미 사용중인 아이디 입니다. -> " + user.getUserId());
+    }
+
+    CoupleEntity coupleEntity = coupleRepository.save(new CoupleEntity());
+
+    user.setCoupleId(coupleEntity.getCoupleId());
+    user.setPassword(this.passwordEncoder.encode(user.getPassword()));
+    user.setCreatedDate(LocalDateTime.now());
+
+    return this.userRepository.save(user.toEntity());
+  }
+
+  /**
+   * 상대방이 이미 가입한 상태인 회원 등록하는 메서드
+   *
+   * @param user (회원에 대한 정보)
+   * @return 저장된 회원 정보
+   */
+  public UserEntity registerWithPartner(SignUpWithPartner user) {
+    boolean exists = this.userRepository.existsByUserId(user.getUserId());
+    if (exists) {
+      throw new RuntimeException("이미 사용중인 아이디 입니다. -> " + user.getUserId());
+    }
```

Comment:
exists를 굳이 할당할 필요가 없을 것 같습니다
```suggestion
    if (this.userRepository.existsByUserId(user.getUserId())) {
      throw new RuntimeException("이미 사용중인 아이디 입니다. -> " + user.getUserId());
    }
```